packages/manager/apps/catalog-revamp/src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,205 @@
+      label: string,
+      event: CustomEvent<{ checked: boolean; value: string }>,
+    ) => {
+      setHasInteracted(true);
+      const { checked } = event.detail;
+      if (type === 'category') {
+        setSelectedCategoriesLocal((prevSelected) =>
+          checked
+            ? [...prevSelected, label]
+            : prevSelected.filter((item) => item !== label),
+        );
+      } else {
+        setSelectedUniversesLocal((prevSelected) =>
+          checked
+            ? [...prevSelected, label]
+            : prevSelected.filter((item) => item !== label),
+        );
+      }
+    };
+
+    const FilterItem: React.FC<{
+      label?: string;
+      count?: number;
+      loading?: boolean;
+      type: 'category' | 'universe';
+    }> = ({ label, count, loading, type }) => {
+      const isChecked =
+        type === 'category'
+          ? selectedCategories.includes(label || '')
+          : selectedUniverses.includes(label || '');
+
+      const resultsNumber = count ? ` (${count})` : '';
+
+      return (
+        <div className="break-inside-avoid">
+          <OsdsCheckbox
+            checked={isChecked}
+            onOdsCheckedChange={(event: CustomEvent) =>
+              handleCheckboxChange(type, label, event)
+            }
+          >
+            <OsdsCheckboxButton
+              size={ODS_CHECKBOX_BUTTON_SIZE.sm}
+              color={ODS_THEME_COLOR_INTENT.primary}
+            >
+              <span slot="end">
+                {loading ? (
+                  <OsdsSkeleton />
+                ) : (
+                  <OsdsText
+                    color={ODS_THEME_COLOR_INTENT.default}
+                    level={ODS_TEXT_LEVEL.button}
+                    size={ODS_TEXT_SIZE._600}
+                  >
+                    {label} {resultsNumber}
+                  </OsdsText>
+                )}
+              </span>
+            </OsdsCheckboxButton>
+          </OsdsCheckbox>
+        </div>
+      );
+    };
+
+    return (
+      <>
+        <div className="filters-container filters-container grid grid-cols-1 md:flex">
+          <div className="filters-universes">
+            <OsdsText
+              level={ODS_TEXT_LEVEL.heading}
+              size={ODS_TEXT_SIZE._400}
+              color={ODS_THEME_COLOR_INTENT.text}
+              className="title"
+            >
+              {t('manager_catalog_filters_universes')}
+            </OsdsText>
+            {universes.map(
+              (item: { universe: string; count: number }, id: number) => (
+                <FilterItem
+                  key={`item-${id}`}
+                  label={item.universe}
+                  count={item.count}
+                  type="universe"
+                />
+              ),
+            )}
+            {!universes.length &&
+              [...Array(5)].map((_, id: number) => (
+                <FilterItem key={`item-${id}`} loading={true} type="universe" />
+              ))}
```

Comment:
```suggestion
            {!universes.length &&
              [...Array(5)].map((_, id: number) => (
                <FilterItem key={`item-${id}`} loading type="universe" />
              ))}
```

Make a component for this for better readability

Reply:
Done

Reply:
Push before resolving comments

Reply:
Alright, will do that for next reviews